### PR TITLE
[IntelOpenMP] Make artifacts lazy

### PR DESCRIPTION
**Pull request by topolarity**
This is consistent with the `lazy_artifacts=true` we have in MKL_jll

### CODE REVIEW

**Comment by staticfloat** [comment]
Looks like they've deleted their win32 builds :/

https://anaconda.org/conda-forge/mkl/files?page=1

**Comment by topolarity** [comment]
Ironically, the Windows machines still have the file in their cache but the Linux machines do not...

**Comment by topolarity** [comment]
OK switching to PyPi seems to have done the trick.

File: I/IntelOpenMP/build_tarballs.jl [hunk 1]
```diff
@@ -1,46 +1,50 @@
...
+unzip -d intel_openmp-$target intel_openmp-$target.whl
+
 if [[ ${target} == *i686-w64-mingw* ]]; then
-    mv ${target}/bin32/* "${libdir}/."
+    mv intel_openmp-${target}/intel_openmp-*.data/data/bin32/* "${libdir}/."
+
+    # These import libraries go inside the actual lib folder, not the bin folder with the DLLs
+    mkdir -p $WORKSPACE/destdir/lib
+    mv intel_openmp-${target}/intel_openmp-*.data/data/lib32/libiomp5md.lib "$WORKSPACE/destdir/lib/"
+    mv intel_openmp-${target}/intel_openmp-*.data/data/lib32/libiompstubs5md.lib "$WORKSPACE/destdir/lib/"
```
Comment (giordano) [inline I/IntelOpenMP/build_tarballs.jl:30]:
```suggestion
    install -Dvm 755 intel_openmp-${target}/intel_openmp-*.data/data/bin32/* -t "${libdir}"

    # These import libraries go inside the actual lib folder, not the bin folder with the DLLs
    install -Dv intel_openmp-${target}/intel_openmp-*.data/data/lib32/libiomp5md.lib -t "${prefix}/lib/"
    install -Dv intel_openmp-${target}/intel_openmp-*.data/data/lib32/libiompstubs5md.lib -t "${prefix}/lib"
```

File: I/IntelOpenMP/build_tarballs.jl [hunk 2]
```diff
@@ -1,46 +1,50 @@
...
 fi
 if [[ ${target} == *x86_64-w64-mingw* ]]; then
-    mv ${target}/Library/bin/* "${libdir}/."
+    mv intel_openmp-${target}/intel_openmp-*.data/data/Library/bin/* "${libdir}/."
```
Comment (giordano) [inline I/IntelOpenMP/build_tarballs.jl:33]:
```suggestion
    install -Dvm 755 intel_openmp-${target}/intel_openmp-*.data/data/Library/bin/* -t "${libdir}"
```

File: I/IntelOpenMP/build_tarballs.jl [hunk 3]
```diff
@@ -1,46 +1,50 @@
...
 
     # These import libraries go inside the actual lib folder, not the bin folder with the DLLs
     mkdir -p $WORKSPACE/destdir/lib
```
Comment (giordano) [inline I/IntelOpenMP/build_tarballs.jl:36]:
```suggestion
```

File: I/IntelOpenMP/build_tarballs.jl [hunk 4]
```diff
@@ -1,46 +1,50 @@
...
-    cp ${target}-dpcpp/Library/lib/libiomp5md.lib "$WORKSPACE/destdir/lib/"
-    cp ${target}-dpcpp/Library/lib/libiompstubs5md.lib "$WORKSPACE/destdir/lib/"
+    mv intel_openmp-${target}/intel_openmp-*.data/data/Library/lib/libiomp5md.lib "$WORKSPACE/destdir/lib/"
+    mv intel_openmp-${target}/intel_openmp-*.data/data/Library/lib/libiompstubs5md.lib "$WORKSPACE/destdir/lib/"
```
Comment (giordano) [inline I/IntelOpenMP/build_tarballs.jl:38]:
```suggestion
    install -Dv intel_openmp-${target}/intel_openmp-*.data/data/Library/lib/libiomp5md.lib -t "${prefix}/lib"
    install -Dv intel_openmp-${target}/intel_openmp-*.data/data/Library/lib/libiompstubs5md.lib -t "${prefix}/lib"
```

File: I/IntelOpenMP/build_tarballs.jl [hunk 5]
```diff
@@ -1,46 +1,50 @@
...
 fi
 if [[ ${target} == *i686-linux-gnu* ]]; then
-    mv ${target}/lib32/* "${libdir}/."
+    mv intel_openmp-${target}/intel_openmp-*.data/data/lib32/* "${libdir}/."
```
Comment (giordano) [inline I/IntelOpenMP/build_tarballs.jl:41]:
```suggestion
    install -Dvm 755 intel_openmp-${target}/intel_openmp-*.data/data/lib32/* -t "${libdir}"
```

File: I/IntelOpenMP/build_tarballs.jl [hunk 6]
```diff
@@ -1,46 +1,50 @@
...
 fi
 if [[ ${target} == *x86_64-linux-gnu* ]]; then
-    mv ${target}/lib/* "${libdir}/."
+    mv intel_openmp-${target}/intel_openmp-*.data/data/lib/* "${libdir}/."
```
Comment (giordano) [inline I/IntelOpenMP/build_tarballs.jl:44]:
```suggestion
    install -Dvm 755 intel_openmp-${target}/intel_openmp-*.data/data/lib/* -t "${libdir}"
```

File: I/IntelOpenMP/build_tarballs.jl [hunk 7]
```diff
@@ -1,46 +1,50 @@
...
+    FileSource("https://files.pythonhosted.org/packages/ed/a6/92e7356c981fcfefa4fbd132791d28656ec02ff62124d283b10219999d5d/intel_openmp-2024.2.1-py2.py3-none-manylinux1_i686.whl",
+               "ba6de86c394331719c795e743bcff3bbc10c3a16a0d3622d90819593de4c7e34"; filename="intel_openmp-i686-linux-gnu.whl"),
+    FileSource("https://files.pythonhosted.org/packages/78/2d/64570ae938a8ee2337ed8ba28ae1d85d3555ee6e5faadabea9e8b43a900d/intel_openmp-2024.2.1-py2.py3-none-manylinux1_x86_64.whl",
+               "21892ea07a9c6e164707d2b44e6f396f9f2ab6cc7e108755011fad59b596182a"; filename="intel_openmp-x86_64-linux-gnu.whl"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir
 mkdir -p "${libdir}"
```
Comment (giordano) [inline I/IntelOpenMP/build_tarballs.jl:21]:
```suggestion
```

**Comment by topolarity** [comment]
Anything else need fixing here @giordano ?

**Pull request merged**